File: templates/node-nestjs/src/health/health.controller.ts
import { Controller, Get } from "@nestjs/common";

@Controller()
export class HealthController {
  @Get("health")
  check() {
    return { status: "ok" };
  }

  // Readiness probe — customize: add database/redis checks for production
  @Get("ready")
  ready() {
    return { status: "ready" };
  }
}

File: examples/java-spring/src/main/java/com/theo/app/controller/HealthController.java
package com.theo.app.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
public class HealthController {

    @GetMapping("/")
    public Map<String, String> root() {
        return Map.of("message", "Hello from Theo!");
    }

    @GetMapping("/health")
    public Map<String, String> health() {
        return Map.of("status", "ok");
    }

    // Readiness probe — customize: add database/redis checks for production
    @GetMapping("/ready")
    public Map<String, String> ready() {
        return Map.of("status", "ready");
    }
}
